commit 78bd3477745f1ada9578a79f61edb41886bec1cb
Author: Stanislav Malyshev <stas@php.net>
Date:   Sat Dec 29 18:25:37 2018 -0800

    Fix bug #77247 (heap buffer overflow in phar_detect_phar_fname_ext)

diff --git a/ext/phar/phar.c b/ext/phar/phar.c
index 82a9ef3194..0d2173195c 100644
--- a/ext/phar/phar.c
+++ b/ext/phar/phar.c
@@ -2008,7 +2008,7 @@ next_extension:
 	}
 
 	while (pos != filename && (*(pos - 1) == '/' || *(pos - 1) == '\0')) {
-		pos = memchr(pos + 1, '.', filename_len - (pos - filename) + 1);
+		pos = memchr(pos + 1, '.', filename_len - (pos - filename) - 1);
 		if (!pos) {
 			return FAILURE;
 		}
diff --git a/ext/phar/tests/bug77247.phpt b/ext/phar/tests/bug77247.phpt
new file mode 100644
index 0000000000..588975f9f2
--- /dev/null
+++ b/ext/phar/tests/bug77247.phpt
@@ -0,0 +1,14 @@
+--TEST--
+PHP bug #77247 (heap buffer overflow in phar_detect_phar_fname_ext)
+--SKIPIF--
+<?php if (!extension_loaded("phar")) die("skip"); ?>
+--FILE--
+<?php
+try {
+var_dump(new Phar('a/.b', 0,'test.phar'));
+} catch(UnexpectedValueException $e) {
+	echo "OK";
+}
+?>
+--EXPECT--
+OK
\ No newline at end of file
